lib/Sources/BluetoothAction/Options+Decode.swift

```diff
@@ -127,13 +127,15 @@ extension [JsType] {
     func compactMapToUUIDs() -> [UUID]? {
         self.compactMap { $0.string.toUuid() }
     }
+}
 
+extension Dictionary where Key == String, Value == JsType {
     func compactMapToUint8Array() -> [UInt8]? {
-        self.compactMap { $0.number }.compactMap { UInt8(truncating: $0) }
+        self.sorted(by: { $0.key < $1.key }).compactMap { $0.value.number }.compactMap { UInt8(truncating: $0) }
```

Comment:
There is probably a much better way to convert from the dictionary to the `[UInt8]` array, feedback would be appreciated.

The major problem I faced / had to solve was: the data ordering in the dictionary isn't guaranteed to be in-order (hence the `sorted` call):

![Screenshot 2025-03-27 at 10 31 46 AM](https://github.com/user-attachments/assets/69a4b614-0183-4f2c-a67b-d35f2006aabe)

Reply:
There are a bunch of ways to do this, but I think sorting and mapping like this is the best option.

A small nit/tweak on the syntax we can pass the initializer directly instead of a lambda on the last step e.g:
```suggestion
        self.sorted(by: { $0.key < $1.key }).compactMap { $0.value.number }.compactMap(UInt8.init(truncating:))
```

and ultra nit, could combine those two steps using `Optional.map` if you want, either way is fine with me though:
```suggestion
        self.sorted(by: { $0.key < $1.key }).compactMap { $0.value.number.map(UInt8.init(truncating:)) }
```